### DISCUSSION TITLE
In search, what is the point of `index.FilterField`?

### DISCUSSION THREAD

**Discussion by ssyberg** [wagtail/wagtail#12178] [Q&A] [answered]
Just looking at this example below from the docs, isn't `.filter` a standard ORM method on a queryset and if so why do you need the filter field declaration at all? I'm sure I'm missing something simple...

```
from wagtail.search import index
from django.utils import timezone

class EventPage(Page):
    description = models.TextField()
    date = models.DateField()

    search_fields = Page.search_fields + [ # Inherit search_fields from Page
        index.SearchField('description'),
        index.FilterField('date'),
    ]


# Get future events which contain the string "Christmas" in the title or description
>>> EventPage.objects.filter(date__gt=timezone.now()).search("Christmas")
```

**Comment by gasman** [comment, marked as answer]
Not all search backends work through ORM queries - for example, the Elasticsearch backend builds its own separate store of data, where the querying and filtering happens. The `FilterField` is necessary so that we know which fields of the original model should be copied across to the Elasticsearch data store.

**Reply by ssyberg** [reply]
Ah thanks for clarifying, I had assumed `filter` was always ORM but sounds like it's being overridden depending on your search backend. That implies that this only happens when it's chained with a `.search` correct? i.e. if I am using the elastic search backend but I'm just doing a `foo.objects.filter()` that is still 100% ORM/database and does not interact with elasticsearch?

**Reply by gasman** [reply]
Yes, that's correct - if your query doesn't contain a `.search` clause then it will be handled as an ordinary ORM query, and the `search_fields` definition won't come into play.
